src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,75 @@
+import { useMemo } from 'react';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import { useLocation } from 'react-router-dom';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { Todo } from '../../types/Todo';
+import { TodoStatus } from '../../types/TodoStatus';
+
+type Props = {
+  todos: Todo[],
+  tempoTodo: Todo | null,
+  loadingTodoIds: number[];
+  handleTodoRemove: (id: number) => void,
+  handleUpdateTodo: (id: number, data: boolean | string) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempoTodo,
+  loadingTodoIds,
+  handleTodoRemove,
+  handleUpdateTodo,
+}) => {
+  const { pathname } = useLocation();
+
+  const filteredTodos = useMemo(() => todos.filter((todo) => {
+    switch (pathname) {
+      case TodoStatus.Active:
+        return !todo.completed;
+      case TodoStatus.Completed:
+        return todo.completed;
+      default:
+        return TodoStatus.All;
+    }
+  }), [todos, pathname]);
```

Comment:
No need to filter in case TodoStatus.All
NOTE: filter callback should return boolean value - you have string in default case